packages/ui/src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import {
+  forwardRef,
+  HTMLChakraProps,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput as ChakraNumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  UseNumberInputProps,
+} from '@chakra-ui/react';
+
+type NumberInputSize = 'sm' | 'md' | 'lg';
+
+type NumberInputMode = 'numeric' | 'decimal';
+
+export interface NumberInputProps extends Omit<HTMLChakraProps<'div'>, keyof UseNumberInputProps> {
+  /**
+   * Determines number input's paddings and height.
+   */
+  size: NumberInputSize;
+  /**
+   * Determines type of data that might be entered by the user (numeric or decimal).
+   */
+  inputMode?: NumberInputMode;
+  /**
+   * Determines if number input is disabled or not .
+   */
+  disabled?: boolean;
+  value?: number;
+  onChange?: (arg: string) => void;
+  max?: number;
+  min?: number;
+  precision?: number;
+  step?: number;
+}
+
+export const lgStyleProps = {
+  height: '48px',
+  padding: '1px 0px 1px 16px',
+};
+
+export const mdStyleProps = {
+  height: '40px',
+  padding: '1px 0px 1px 16px',
+};
+
+export const smStyleProps = {
+  height: '32px',
+  padding: '1px 0px 1px 12px',
+};
+
+export const NumberInput = forwardRef<NumberInputProps, 'input'>(
+  (
+    {
+      size,
+      inputMode = 'decimal',
+      disabled = false,
+      'aria-describedby': ariaDescribedby,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledby,
+      value,
+      name,
+      max,
+      min,
+      precision,
+      onChange,
+      step,
+    },
+    ref,
+  ) => {
+    const styleProps = {
+      borderRadius: '6px',
+      width: '200px',
+    };
+
+    const sizeProps = {
+      sm: smStyleProps,
+      md: mdStyleProps,
+      lg: lgStyleProps,
+    };
+
+    const hoverStyleProps = {
+      border: '1px solid #9CA3AF',
+    };
+
+    const focusStyleProps = {
+      'box-shadow': '0px 0px 0px 3px rgba(56, 189, 248, 0.6)',
+    };
+
+    const invalidStyleProps = {
+      border: '2px solid #EF4444',
+    };
```

Comment:
All those styles probably could be moved to the theme styling of the NumberInput